emsdk.py

```diff
@@ -2643,10 +2651,26 @@ def get_env_vars_to_add(tools_to_activate, system, user):
   for tool in tools_to_activate:
     config = tool.activated_config()
     if 'EMSCRIPTEN_ROOT' in config:
-      # For older emscripten versions that don't use this default we export
-      # EM_CACHE.
-      em_cache_dir = os.path.join(config['EMSCRIPTEN_ROOT'], 'cache')
-      env_vars_to_add += [('EM_CACHE', em_cache_dir)]
+      # For older emscripten versions that don't use an embedded cache by
+      # default we need to export EM_CACHE.
+      #
+      # Sadly, we can't put this in the config file since those older versions
+      # also didn't read the `CACHE` key from the config file:
+      #
+      # History:
+      # - 'CACHE' config started being honored in 1.39.16
+      #   https://github.com/emscripten-core/emscripten/pull/11091
+      # - Default to embedded cache started in 1.39.16
+      #   https://github.com/emscripten-core/emscripten/pull/#11126
+      #
+      # Since setting EM_CACHE in the environment effects the entire machine
+      # we want to avoid this except when installing these older emscripten
+      # versions that really need it.
+      version = parse_emscripten_version(config['EMSCRIPTEN_ROOT'])
+      if version <= [1, 39, 16]:
```

Comment:
```suggestion
      if version < [1, 39, 16]:
```
Seems like we need this in earlier versions, not this exact one?